src/models/regression/index.ts

```diff
@@ -1,9 +1,77 @@
 import { RandomForestRegression } from "ml-random-forest";
 import MultivariateLinearRegression from "ml-regression-multivariate-linear";
-import { RobustPolynomialRegression } from "ml-regression-robust-polynomial";
 import { kombi } from "kombi";
 import data from "./WineQualityDataSetRed.json";
 
-console.log(data);
-console.log(RandomForestRegression, MultivariateLinearRegression, RobustPolynomialRegression, kombi);
+const rootMeanSquareError = (predicted: number[], observed: number[]) => {
+  let sum = 0;
+  for (let i = 0; i < predicted.length; i++) {
+    const squareError = Math.pow(predicted[i] - observed[i], 2);
+    sum += squareError;
+  }
+  return Math.sqrt(sum / predicted.length);
+}
 
+const calculateSuccessPercentage = (predicted: number[], observed: number[]) => {
+  let successCount = 0;
+  for (let i = 0; i < predicted.length; i++) {
+    const squareError = predicted[i] - observed[i];
+    if ( squareError === 0){
+      successCount++;
+    }
+  }
+  return (successCount / predicted.length) * 100;
+}
+
+const inputs = data.map((el) => Object.values(el).slice(0,-1));
+const outputs = data.map((el) => el.quality);
+const TRAINING_LENGTH = data.length * 95 / 100; // 95 % of the whole data
+const trainX = inputs.slice(0, TRAINING_LENGTH);
+const trainY = outputs.slice(0, TRAINING_LENGTH);
+const predictX = inputs.slice(TRAINING_LENGTH);
+const predictY = outputs.slice(TRAINING_LENGTH);
```

Comment:
See if you can use this library (https://github.com/MarcoABCardoso/kfold-experiment#readme) for cross-validation and for generating confusion matrix